homeassistant/components/mqtt/config_flow.py

```diff
@@ -704,7 +643,7 @@ async def _async_validate_broker_settings(
             default=verification_mode,
         )
     ] = BROKER_VERIFICATION_SELECTOR
-    if current_ca_certificate is None and verification_mode == "custom":
+    if current_ca_certificate is not None or verification_mode == "custom":
```

Comment:
We want to be able to upload ca cert and key if the option certificate validation option is set to `custom` or if a custom ca cert is already set up.